packages/language/src/validation/messages/error-severity/IBM2412I-IBM2410I-IBM2409I-handle-return-stmt-and-returns-att.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { ValidationAcceptor } from "../../validator";
+import * as AST from "../../../syntax-tree/ast";
+import { diagnosticFromCode } from "../../../language-server/types";
+import * as PLICodes from "../pli-codes";
+import {
+  forEachNode,
+  TraversalState,
+  traverseAllNodes,
+} from "../../../syntax-tree/ast-iterator";
+
+/**
+ * IBM2412I: If a procedure contains a RETURN statement, it should have the RETURNS attribute
+ * specified on its PROCEDURE statement.
+ *
+ * IBM2409I: RETURN statement without an expression is invalid inside a nested PROCEDURE that
+ * specifies the RETURNS attribute. All RETURN statements inside functions must specify a value
+ * to be returned.
+ *
+ * IBM2410I: Functions must contain at least one RETURN statement.
+ *
+ * @param node The AST node being analyzed.
+ * @param acceptor The mechanism used to collect validation issues.
+ * @returns Validation results or diagnostics, as appropriate.
+ */
+export function IBM2412I_IBM2410I_IBM2409I_handle_return_stmt_and_returns_att(
+  node: AST.ProcedureStatement,
+  acceptor: ValidationAcceptor,
+): void {
+  const returnStmts: AST.ReturnStatement[] = [];
+
+  const traverseChildren = (node: AST.SyntaxNode) => {
+    traverseAllNodes(node, (n) => {
+      if (n.kind === AST.SyntaxKind.ProcedureStatement)
+        return TraversalState.Skip;
+      if (n.kind === AST.SyntaxKind.ReturnStatement)
+        returnStmts.push(n as AST.ReturnStatement);
+      return TraversalState.Continue;
+    });
+  };
+  forEachNode(node, (child) => {
+    traverseChildren(child);
+  });
```

Comment:
Minor: We can rewrite this to make this a bit easier to read.
```suggestion
  traverseAllNodes(node, (n) => {
    if (n !== node && n.kind === AST.SyntaxKind.ProcedureStatement)
      return TraversalState.Skip;
    if (n.kind === AST.SyntaxKind.ReturnStatement)
      returnStmts.push(n as AST.ReturnStatement);
    return TraversalState.Continue;
  });
```

Reply:
Thanks, it looks cleaner indeed!